### ISSUE TITLE
Sincerely ask for help

### ISSUE THREAD

**Issue by Tomasyao77** [vicwer/C3AE_Age_Estimation#22]
Hello.Are you the author of C3AE, or do you just reproduce their code according to this paper.

I am confused about some places in the article.In the article, it is said that "Our results on with/without pretrained process are very similar. "

In fact, I had a poor result in training on imdb-wiki and then on MORPH2.If I only train on MORPH2, the average MAE on test dataset is difficult to reach 2.75.The average MAE of mine on test dataset is nearly 3.7.

I really want to know the reason, because there are many papers that are difficult to reproduce the original results, whether it is my wrong operation.Thank you very much.

**Comment by vicwer** [comment]
sorry, I am not the author. In fact, these papers always talk big.If you want to reproduce the original results, you need have the skill of adjusting parameters of NewBee.
I code because of his network structure.

**Comment by Tomasyao77** [comment]
OK, thank you for opening up this code and saving us a lot of work. I will continue to research, learn and debug. Thank you again

**Comment by vicwer** [comment]
You are welcome.
Good luck~

**Comment by vicwer** [comment]
@Tomasyao77 Hi, is your loss nan when you train the network by my codes ?

**Comment by Tomasyao77** [comment]
I have met one time.But I don't remember clearly.Now the code is running good.
Probably the reason of why loss nan is that l1 loss or kl loss function is not right.
After consulting you several times before, my code is like this：

```
def l1_loss(preds, labels):
    preds = tf.reshape(preds, (-1,))
    labels = tf.reshape(labels, (-1,))
    l1_loss = tf.abs(preds - labels)
    return tf.reduce_mean(l1_loss)

def kl_loss(preds, labels, l1):
    preds = tf.nn.softmax(preds)
    kl_loss = tf.reduce_sum(labels * tf.log(preds + 1e-10), axis=1)
    kl_loss = tf.reshape(kl_loss, (cfg.batch_size,))
    _, k_idx = tf.nn.top_k(kl_loss, tf.cast(cfg.batch_size * cfg.ohem_ratio, tf.int32))
    loss = tf.gather(kl_loss, k_idx)
    return -tf.reduce_mean(loss) + l1
```

**Comment by vicwer** [comment]
ok, thanks.
